src/integrations/magni.ts

```diff
@@ -21,3 +28,36 @@ export const postFeedback = async (
       'Content-Type': 'application/json',
     },
   });
+
+  if (!res.ok) throw new ValidationError(await res.text());
+};
+
+interface SessionResponse {
+  sessions: SearchSession[];
+}
+
+export interface SearchSessionParams {
+  limit?: number;
+  lastId?: string;
+}
+
+export const getSessions = async (
+  userId: string,
+  { limit = 30, lastId }: SearchSessionParams = {},
+): Promise<SearchSession[]> => {
+  const params = new URLSearchParams({ limit: limit.toString() });
+
+  if (lastId) params.append('lastId', lastId);
+
+  const url = `${magniOrigin}/sessions?${params.toString()}`;
+  const res = await fetch(url, {
+    ...fetchOptions,
+    headers: { 'X-User-Id': userId },
```

Comment:
Don't you miss here the content type header?

Reply:
We should not need it here as we are not passing anything to `body` to define its type.